src/app/modules/color-scheme/color-bar/color-bar.component.ts

```diff
@@ -0,0 +1,71 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { ColorScheme } from '../color-schemes';
+
+/**
+ * Color bar component for the color scheme popup and color scheme selector
+ */
+@Component({
+  selector: 'ccf-color-bar',
+  templateUrl: './color-bar.component.html',
+  styleUrls: ['./color-bar.component.scss']
+})
+export class ColorBarComponent implements OnInit {
+
+  /**
+   * Color scheme to be displayed in the color bar
+   */
+  @Input() colorScheme: ColorScheme;
+
+  @Input() selected = false;
+
+  /**
+   * Emits the newly selected color
+   */
+  @Output() colorChange = new EventEmitter<string | undefined>();
+
+  /**
+   * Enables dynamic styling for gradient bars
+   */
+  gradientstyle: string;
```

Comment:
Camel case! Also a good idea is to change this into a getter that creates the string on demand. Would also allow it to automatically update when the colorScheme input changes.